src/layers/recurrent.jl

```diff
@@ -157,12 +170,9 @@ end
 function (m::LSTMCell{A,V,<:NTuple{2,AbstractMatrix{T}}})((h, c), x::Union{AbstractVecOrMat{T},OneHotArray}) where {A,V,T}
   b, o = m.b, size(h, 1)
   g = m.Wi*x .+ m.Wh*h .+ b
-  input = σ.(gate(g, o, 1))
-  forget = σ.(gate(g, o, 2))
-  cell = tanh.(gate(g, o, 3))
-  output = σ.(gate(g, o, 4))
-  c = forget .* c .+ input .* cell
-  h′ = output .* tanh.(c)
+  input, forget, cell, output = multigate(g, o, Val(4))
+  c = @. σ(forget) * c + σ(input) * tanh(cell)
+  h′ = @. σ(output) * tanh(c)
   sz = size(x)
   return (h′, c), reshape(h′, :, sz[2:end]...)
```

Comment:
Since the new `h` gets a prime, probably the new `c` should too?
```suggestion
  c′ = @. σ(forget) * c + σ(input) * tanh(cell)
  h′ = @. σ(output) * tanh(c′)
  return (h′, c′), reshape(h′, :, size(x)[2:end]...)
```
Not too sure what the reshape is doing, is it `x isa AbstractVector ? vec(h′) : h′`? Might be worth pulling out to a tiny function, from all of these.

Reply:
Good idea, I've now done both.